src/printer_tree.c

```diff
@@ -4337,3 +4418,131 @@ tree_print_parsed_submodule(struct ly_out *out, const struct lysp_submodule *sub
 
     return erc;
 }
+
+/* @param ext[in] context allocated for extension */
+/* @param refs[out] set of lysc nodes matching parent-refernce xpaths? */
+static LY_ERR
+trb_parent_references(const struct lysc_ext_instance *ext, struct ly_set **refs)
+{
+    LY_ERR res;
+    struct ly_set *pref_set = NULL;
+    struct ly_set *snode_set;
+    struct ly_set *results_set = NULL;
+    struct lyd_node *ext_data;
+    ly_bool ext_data_free;
+
+    /* get operational data with ietf-yang-library and ietf-yang-schema-mount data */
+    if ((res = lyplg_ext_get_data(ext->module->ctx, ext, (void **)&ext_data, &ext_data_free))) {
+        return res;
+    }
+
+    LY_CHECK_GOTO(res = lyd_find_xpath(ext_data, "/ietf-yang-schema-mount:schema-mounts/mount-point/shared-schema/parent-reference",
+                                       &pref_set), out);
+    if (pref_set->count == 0) {
+        goto out;
+    }
+
+    LY_CHECK_GOTO(res = ly_set_new(&results_set), out);
+
+    for (uint32_t i = 0; i < pref_set->count; ++i) {
+        struct lyd_node_term *term;
+        struct lyd_value_xpath10 *xp_val;
+        char *value;
+        struct ly_err_item *err;
+
+        term = (struct lyd_node_term *)pref_set->dnodes[i];
+        LYD_VALUE_GET(&term->value, xp_val);
+        LY_CHECK_GOTO(res = lyplg_type_print_xpath10_value(xp_val, LY_VALUE_JSON, NULL, &value, &err), out);
+        LY_CHECK_ERR_GOTO(res = lys_find_xpath(ext->module->ctx, NULL, value, 0, &snode_set), free(value), out);
+        free(value);
+        for (uint32_t sn = 0; sn < snode_set->count; sn++) {
+            struct lysc_node *top = snode_set->snodes[sn];
+
+            while (top->parent)
```

Comment:
And in a case like this:
```
<parent-reference>/if:interfaces/if:interface/if:name</parent-reference>
<parent-reference>/if:interfaces/if:interface/ip:ipv4/ip:address/ip:ip</parent-reference>
```
you would expect a single trimmed tree containing both name and ip?  That's probably too simple of an example, but I think we could have some awkward diagrams.  The tree diagram RFC and the network instance RFC (8529) both show these parent reference top-level nodes with collapsed trees ("...").   Maybe it is safer to mimic this behavior since there are examples of it?

Reply:
> show these parent reference top-level nodes with collapsed trees ("...")

I think `...` was meant to just be a simplification so the example does not have to have the whole schema trees printed, they are supposed to be there. In any case, printing the whole schema trees is misleading because they are not actually accessible so it does not make sense to me.

> you would expect a single trimmed tree containing both name and ip?

Exactly but when we thought about it, "merging" the paths is a non-trivial problem that we wanted to solve later ourselves although you are free to give it a shot yourself. For now we would be fine with printing the path of each parent reference separately.

Reply:
For now each parent reference is printed separately.